test/lib/smartSurveyAPICall.test.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+const proxy = require('proxyquire');
+const sinon = require('sinon');  
+const chai = require('chai').use(require('sinon-chai'));
+const should = chai.should();
+const proxyquire = require('proxyquire');
+let stubAxios = {};
+
+// Proxyquire syntax: first argument: point to your module (which holds your dependency)
+// 2nd argument: An object with the key: module name to proxy & the value the empty object above
+// So this means smartSurveyAxiosAPI will now point to this empty object, i.e. a stub
+const smartSurveyAxiosAPI = proxyquire('../../lib/smartSurveyAPICall', {
+  'axios': stubAxios,
+});
+
+// create an empty function using echma 5 syntax because of using the this context
+var stubSmartSurvey = function(params) {
```

Comment:
`function () {}` is still ECMA 6 syntax, it's just not an arrow function `() => {}` which as you say won't have a `this` context. and cannot be used for describing constructors which is what he's doing here.
https://www.sitepoint.com/es6-arrow-functions-new-fat-concise-syntax-javascript/
covers it reasonably well